### PR TITLE
Rek approve reimbursement swag hand

### PR DESCRIPTION
**Pull request by kilbergr**
## Description
This is a PR that includes the following work:
* New Approve Reimbursement handler
* Test for new approve reimbursement handler
* New approve reimbursement swagger endpoint
* Payment panel loading information from the loaded PPM's advance
* Payment panel loading approveReimbursement ACTION from ducks
* Creation of approveReimbursement ACTION in ducks
* testdatagen now includes MakeReimbursement
* Only works for a single reimbursement/advance

## Code Review Verification Steps

* [x] All tests pass.
* [x] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [x] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [x] There are no aXe warnings for UI.
* [x] This works in IE.
* [x] Request review from a member of a different team.
* [x] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/157748113) for this change
* [Pivotal story](https://www.pivotaltracker.com/story/show/157748853) for this change
screen shot 2018-05-30 at 11 47 32 am
screen shot 2018-05-30 at 11 47 39 am
screen shot 2018-05-30 at 3 02 50 pm

### CODE REVIEW

File: pkg/handlers/office.go [hunk 1]
```diff
@@ -71,3 +71,33 @@ func (h ApprovePPMHandler) Handle(params officeop.ApprovePPMParams) middleware.R
...
+
+	// #nosec UUID is pattern matched by swagger and will be ok
+	reimbursementID, _ := uuid.FromString(params.ReimbursementID.String())
+
+	reimbursement, err := models.FetchReimbursement(h.db, session, reimbursementID)
+	if err != nil {
+		return responseForError(h.logger, err)
+	}
+
+	reimbursement.Status = models.ReimbursementStatusAPPROVED
```
Comment (macrael) [inline pkg/handlers/office.go]: Please use the state machine methods defined on the reimbursements model for this. You should call reimbursement.Approve() instead.

You will then need to handle the error that throws. e.g. this API should return an error if you attempt to approve a reimbursement that has already been approved or paid.

File: pkg/models/reimbursement.go [hunk 2]
```diff
@@ -9,6 +9,8 @@ import (
 	"github.com/gobuffalo/validate/validators"
 	"github.com/pkg/errors"
 
+	// "github.com/transcom/mymove/pkg/app"
```
Comment (jim) [inline pkg/models/reimbursement.go]: (nit) remove commented out code.

File: pkg/models/reimbursement.go [hunk 3]
```diff
@@ -18,6 +20,8 @@ type MethodOfReceipt string
 const (
 	// MethodOfReceiptMILPAY captures enum value MIL_PAY
 	MethodOfReceiptMILPAY MethodOfReceipt = "MIL_PAY"
+	// MethodOfReceiptDIRECTDEPOSIT captures enum value DIRECTDESPOSIT
+	MethodOfReceiptDIRECTDEPOSIT = "DIRECT_DEPOSIT"
```
Comment (macrael) [inline pkg/models/reimbursement.go]: I'm confused by this addition. Kim said that there were only three options:

“MilPay”
“GTCC”
“Other account”

I think that both MilPay and Other Account are actually direct deposit. what does Direct Deposit mean?

File: swagger/internal.yaml [hunk 4]
```diff
@@ -2210,6 +2210,33 @@ paths:
           description: request requires user authentication
         403:
           description: user is not authorized
+  /reimbursement/{reimbursementId}/approve:
```
Comment (macrael) [inline swagger/internal.yaml]: If DIRECT_DEPOSIT is to be added, it needs to be added as a choice for ReimbursementMethod in internal.yaml

File: pkg/models/reimbursement.go (hunk 3 above)
Comment (kilbergr) [reply pkg/models/reimbursement.go]: You're right. I thought this was DD for Other but there's another Other below. I'm going to remove this one.

**Pull request merged**